src/features/locators/reducers/addCustomLocator.thunk.ts

```diff
@@ -0,0 +1,71 @@
+import { ActionReducerMapBuilder, createAsyncThunk } from "@reduxjs/toolkit";
+import { Locator, LocatorsState, ValidationStatus } from "../types/locator.types";
+import { getElementFullXpath, parseElementFromString, generateId } from "../../../common/utils/helpers";
+import { addLocatorToPageObj } from "../../pageObjects/pageObject.slice";
+import { addLocators, setScrollToLocator } from "../locators.slice";
+import { getLocatorValidationStatus, evaluateXpath } from "../utils/utils";
+import { PageObjectId } from "../../pageObjects/types/pageObjectSlice.types";
+import { sendMessage } from "../../../pageServices/connector";
+import { locatorsAdapter } from "../locators.selectors";
+
+export const addCustomLocator = createAsyncThunk(
+  "locators/addCustomLocator",
+  async (payload: { newLocator: Locator; pageObjectId: PageObjectId }, thunkAPI) => {
+    let { newLocator } = payload;
+    const { pageObjectId } = payload;
+
+    let { element_id } = newLocator;
+    const { message, locator } = newLocator;
+
+    element_id = element_id || `${generateId()}_${pageObjectId}`;
+
+    if (getLocatorValidationStatus(message) === ValidationStatus.SUCCESS) {
+      try {
+        let { foundHash } = JSON.parse(await evaluateXpath(locator.customXpath!, element_id));
+        const { foundElement } = JSON.parse(await evaluateXpath(locator.customXpath!, element_id));
```

Comment:
Why can't we get `foundElement` together with `foundHash` on line 24?

Reply:
let and const constraints